src/utils/NucleicAcidSearchEngine.cpp

```diff
@@ -621,9 +627,20 @@ class NucleicAcidSearchEngine :
       }
 
       // deisotope
-      Int coef = negative_mode ? -1 : 1;
       // @TODO: what happens here if "precursor_charge" is zero?
-      deisotopeAndSingleChargeMSSpectrum_(spec, coef, coef * precursor_charge, fragment_mass_tolerance, fragment_mass_tolerance_unit_ppm, false, 3, 20, single_charge_spectra);
+      // Check that fragment charge parameters match polarity of parent
+
+      Int base_charge = negative_mode ? -1 : 1;
+      // Check that fragment charge parameters match polarity of parent
+      if ((precursor_charge * base_charge * frag_max_charge < 0 )|| (precursor_charge * base_charge * frag_min_charge <0))
+      {
+        OPENMS_LOG_ERROR
+              << "Error: Precursor charge state opposite polarity from specified fragment charge state range '"
+              << endl;
+      }
+ 
+      Int temp_frag_max_charge = (abs(frag_max_charge)>abs(precursor_charge)) ? precursor_charge * base_charge : temp_frag_max_charge;
+      deisotopeAndSingleChargeMSSpectrum_(spec, frag_min_charge, temp_frag_max_charge, fragment_mass_tolerance, fragment_mass_tolerance_unit_ppm, deisotope_spectra, 3, 20, single_charge_spectra);
```

Comment:
does this support negative charges? I think this needs to be abs

Reply:
Nope, it handles them properly